frontend/src/pages/modelCatalog/screens/RegisterCatalogModel.tsx

```diff
@@ -68,10 +72,13 @@ const RegisterCatalogModel: React.FC = () => {
   const isModelNameValid = isNameValid(formData.modelName);
   const eventName = 'Catalog Model Registered';
 
+  const isModelNameDuplicate = isModelNameExisting(formData.modelName, registeredModels);
+  const hasModelNameError = !isModelNameValid || isModelNameDuplicate;
+
   // passing registeredModels as [] is temporary and will handle this as a part of https://issues.redhat.com/browse/RHOAIENG-20564
   const isSubmitDisabled =
     isSubmitting ||
-    isRegisterCatalogModelSubmitDisabled(formData, {
+    isRegisterCatalogModelSubmitDisabled(formData, isModelNameDuplicate, {
```

Comment:
Could you please remove this comment mentioning your issue here.

Also, just pass the registeredModels to 
`isRegisterCatalogModelSubmitDisabled(formData, registeredModels)`
 this will handle the duplicate error case.

Reply:
Good catch, i've missed that comment.